### PR TITLE
fix: #151, materials now update meshes on uniform setting. [#95480122]

### PR DESCRIPTION
**Pull request by adnan-wahab**

### CODE REVIEW

File: webgl-materials/Material.js [hunk 1]
```diff
@@ -178,6 +177,12 @@ function Material(name, chunk, inputs, options) {
     this.varyings = options.varyings;
     this.attributes = options.attributes;
 
+    this.meshes = [];
+
+    var self = this;
```
Comment (alexanderGugel) [inline webgl-materials/Material.js]: rename to _this please. Linter is going to fail here.

File: webgl-renderables/Mesh.js [hunk 2]
```diff
@@ -651,3 +657,20 @@ Mesh.prototype.draw = function draw () {
 };
 
 module.exports = Mesh;
```
Comment (alexanderGugel) [inline webgl-renderables/Mesh.js]: can you move this to the end of the file?

File: webgl-renderables/Mesh.js [hunk 3]
```diff
@@ -651,3 +657,20 @@ Mesh.prototype.draw = function draw () {
...
+
+function addMeshToMaterial(mesh, material) {
+    if (material.meshes.indexOf(mesh) == -1)
+        material.meshes.push(mesh);
+}
+
+function _traverse(material, callback) {
+	var inputs = material.inputs;
```
Comment (alexanderGugel) [inline webgl-renderables/Mesh.js]: indent

File: webgl-renderables/Mesh.js [hunk 4]
```diff
@@ -651,3 +657,20 @@ Mesh.prototype.draw = function draw () {
 };
 
 module.exports = Mesh;
+
+function addMeshToMaterial(mesh, material) {
+    if (material.meshes.indexOf(mesh) == -1)
```
Comment (alexanderGugel) [inline webgl-renderables/Mesh.js]: `===`

File: webgl-materials/Material.js [hunk 5]
```diff
@@ -194,6 +199,10 @@ Material.prototype.setUniform = function setUniform(name, value) {
     this.uniforms[name] = value;
 
     this.invalidations.push(name);
+
+    this.meshes.forEach(function (mesh) {
```
Comment (alexanderGugel) [inline webgl-materials/Material.js]: Can we change this to a `for` loop?

**Comment by michaelobriena** [comment]
@adnan-wahab do you have an example with and without this fix?

@redwoodfavorite @FarhadG can you review?

**Comment by adnan-wahab** [comment]
before: https://api-te.famo.us/codemanager/v1/containers/0e578483-207f-44ae-8ac8-3762962dfa31/share

after: https://api-te.famo.us/codemanager/v1/containers/0e578483-207f-44ae-8ac8-3762962dfa31/share

File: webgl-renderers/compileMaterial.js [hunk 6]
```diff
@@ -66,6 +66,7 @@ function compileMaterial(material, textureSlot) {
         if (node.texture) textures.push(node.texture);
     });
 
+    console.log(uniforms);
```
Comment (redwoodfavorite) [inline webgl-renderers/compileMaterial.js]: ...

File: webgl-renderers/WebGLRenderer.js [hunk 7]
```diff
@@ -107,7 +107,7 @@ function WebGLRenderer(canvas, compositor) {
     };
 
     this.resolutionName = ['u_resolution'];
-    this.resolutionValues = [];
+    this.resolutionValues = [0, 0];
```
Comment (redwoodfavorite) [inline webgl-renderers/WebGLRenderer.js]: ?

**Comment by michaelobriena** [comment]
@adnan-wahab you have two different PR's on the same branch.  Please close this and make a feautre branch for each.

**Pull request closed without merging**